File: recursive_factorial.java
class abc
{

    static int factorial(int n)
    {
        if (n == 0)
            return 1;

        return n*factorial(n-1);
    }

    public static void main(String[] args)
    {
        int num = 5;
        System.out.println("Factorial of "+ num + " is " + factorial(4));
    }
}

File: factorial/recursive_factorial.c
#include <stdio.h>

long factorial(long n) {
    if (n == 0) return 1;
    return n * factorial(n - 1);
}

int main() {
    long num = 0;
    scanf("%li", &num);

    printf("%li\n", factorial(num));

    return 0;
}